src/MangaJet/app/src/main/java/com/mangajet/mangajet/MainActivity.kt

```diff
@@ -152,21 +157,44 @@ class MainActivity : AppCompatActivity(), ActivityResultCallback<Map<String, Boo
     // Functions that checks report for crash => sends it to email
     private fun checkForCrash(report : String){
 
-        if(report.isEmpty())
-            return
         val email = "mangajetmailbot@gmail.com"
+        val tmpFileName = "backupToMail.zip"
 
+        //create zip file
+        var file : File? = null
+        try {
+            // To Download pages, because it won't be in archive
+            if(StorageManager.isExist(tmpFileName, StorageManager.FileType.DownloadedPages))
+                StorageManager.getFile(tmpFileName, StorageManager.FileType.DownloadedPages).delete()
+            if(report.isEmpty()) // Ask me about this later, Vova
```

Comment:
I think I got it.. Some times report is exist but empty. It isn`t a bad thing, you just must check it at start of function.
But Archive you should create only if user wanted to create it. So move archive creation to Dialog "positive" callback